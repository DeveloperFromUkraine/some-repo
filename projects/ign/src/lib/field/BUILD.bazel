load("@angular//:index.bzl", "ng_module")
load("@io_bazel_rules_sass//:defs.bzl", "sass_binary")

package(default_visibility = ["//visibility:public"])

sass_binary(
    name = "style",
    src = "field.component.scss",
)

ng_module(
    name = "field-module",
    srcs = [
        "field.module.ts",
        "editable-field.component.ts",
        "editable-field-inline.component.ts",
        "field.component.sandbox.ts",
        "field.component.ts",
        "inline-field.component.ts"
    ],
    assets = [
        ":field.component.css",
        "editable-field.component.html",
        "editable-field-inline.component.html",
        "field.component.html",
        "inline-field.component.html"
              ],
    tsconfig = "//:tsconfig.json",
    deps = [
        "@npm//tslib",
        "@npm//@types",
        "@npm//angular-playground",
        "@angular//packages/core",
        "@angular//packages/common",
        "@angular_material//src/lib:material",
        "//projects/ign/src/lib/localization:localization-translation-module",
    ]
)
