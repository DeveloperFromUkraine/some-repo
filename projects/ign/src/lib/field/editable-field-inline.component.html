<div class="label-container">
  <div da="fieldValue"
       id="editable-value"
       tabindex="0"
       [attr.da]="da"
       class="value flex-lable-value">
    <span class="label">{{label}}</span>
    <ng-content></ng-content>
  </div>
  <button aria-describedby="editable-value"
          [attr.da]="daButton"
          [attr.aria-label]="ariaBtnLabel"
          mat-icon-button class="mat-icon-button"
          (click)=handleClick()>
    <mat-icon>{{editIcon}}</mat-icon>
  </button>
</div>