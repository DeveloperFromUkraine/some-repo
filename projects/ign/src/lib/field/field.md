*** HTML ***
# Field

## Use
This component displays a view field, consisting of a label and a value. 

## Sample

<mat-tab-group>
    <mat-tab label="Component Sample">
        <div class="tab-height">
            <ign-field label="The Answer">42</ign-field>
        </div></mat-tab>
    <mat-tab label="HTML"><div class="tab-height">
        <table style="width:100%">
            &lt;ign-field label="The Answer"&gt;42&lt;/ign-field&gt;
        </table>
    </div></mat-tab>
</mat-tab-group>

## Content

* The contents of the tag will be displayed as the value.

## Style

* A label must be provided.
* A value pertaining to the label must be provided if user input is not requested.

## Accessibility

* For editable field and editable inline field components, a default aria label has been added for the edit button.
* For a more descriptive label, add an ariaBtnLabel attribute and provide a more descriptive value.
* Refer to the sample html in the accordion below for examples of usage.

## Field Components

<mat-accordion id="accordion">
    <mat-expansion-panel>
        <mat-expansion-panel-header>
            <mat-panel-title>
                ign-field-inline
            </mat-panel-title>
            <mat-panel-description><i>Field component with inline formatting.</i></mat-panel-description>
        </mat-expansion-panel-header>
        <mat-tab-group>
            <mat-tab label="Component Styling"><div class="tab-height">
                <ul>
                    <li>This component is similar to the ign-field but slightly different look and feel. </li>
                    <li>Label will appear first followed by the text value on the same row.</li>
                    <li>A label must be provided. </li>
                    <li>The contents of the tag will be displayed as the value.</li>
                </ul>
            </div></mat-tab>
            <mat-tab label="HTML"><div class="tab-height">
                <p> &lt;ign-field-inline label="The Answer"&gt;42&lt;/ign-field-inline&gt;</p>
            </div></mat-tab>
        </mat-tab-group>
    </mat-expansion-panel>
    <mat-expansion-panel>
        <mat-expansion-panel-header>
            <mat-panel-title>
                ign-editable-field
            </mat-panel-title>
            <mat-panel-description><i>Editable field component.</i></mat-panel-description>
        </mat-expansion-panel-header>
        <mat-tab-group>
            <mat-tab label="Component Styling"><div class="tab-height">
                <ul>
                    <li>Similar to the field component but also includes an edit button that emits an event that can be subscribed to.</li>
                    <li>A label must be provided. The contents of the tag will be displayed as the value.</li>
                    <li>An editIcon can be optionaly provided; if none passed the button will display mat-icon mode_edit as default. </li>
                </ul>
            </div></mat-tab>
            <mat-tab label="HTML"><div class="tab-height">
                <p> &lt;ign-editable-field label="The Answer" ariaBtnLabel="The Answer" (onIconClick)="myFn($event)"&gt;42&lt;/ign-editable-field&gt;</p>
            </div></mat-tab>
        </mat-tab-group>
    </mat-expansion-panel>
    <mat-expansion-panel>
        <mat-expansion-panel-header>
            <mat-panel-title>
                ign-editable-field-inline
            </mat-panel-title>
            <mat-panel-description><i>Editable field component with inline styling.</i></mat-panel-description>
        </mat-expansion-panel-header>
        <mat-tab-group>
            <mat-tab label="Component Styling"><div class="tab-height">
                <ul>
                    <li>Similar to the editable field above but slightly different look and feel. </li>
                    <li>Label will appear first followed by the text value followed by the edit button on the same row.</li>
                    <li>A label must be provided.</li>
                    <li>The contents of the tag will be displayed as the value.</li>
                </ul>
            </div></mat-tab>
            <mat-tab label="HTML"><div class="tab-height">
                <p> &lt;ign-editable-field-inline label="The Answer" ariaBtnLabel="The Answer" editIcon="home" (onIconClick)="myFn($event)"&gt;42&lt;/ign-editable-field-inline&gt;</p>
            </div></mat-tab>
        </mat-tab-group>
    </mat-expansion-panel>
</mat-accordion>

*** TypeScript *** 

*** ExternalTypeScript ***

*** Imports ***