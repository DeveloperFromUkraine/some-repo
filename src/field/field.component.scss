$field-label-color: #757575;
$field-label-font-size: 0.85rem;

:host {
  display: block;
  margin: 0 0 1.75rem 0;
  line-height: 1.4rem;
}

.label {
  color: $field-label-color;
  font-size: $field-label-font-size;
}

.value {
  word-wrap: break-word;
}

.flex-lable-value {
  flex-grow: 2;
}

.label-container {
  display: flex;
  justify-content: space-between;
}

.mat-icon-button {
  height: 24px;
  width: 24px;
  line-height: normal;
}
