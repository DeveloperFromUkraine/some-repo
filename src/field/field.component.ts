import { Component, Input, Output, EventEmitter } from '@angular/core';

@Component({
  selector: 'ign-field',
  template: `
    <div da="fieldValue" class="value"><ng-content></ng-content></div>
    <div class="label">{{label}}</div>
  `,
  styleUrls: ['field.component.scss'],
})
export class FieldComponent {
  @Input() label: string;
}

@Component({
  selector: 'ign-field-inline',
  template: `
    <div da="fieldValue" class="value">
    <span class="label">{{label}}</span>
    <ng-content></ng-content>
    </div>
  `,
  styleUrls: ['field.component.scss'],
})
export class FieldInlineComponent extends FieldComponent {}
@Component({
  selector: 'ign-editable-field',
  template: `
    <div class="label-container">
    <div [attr.da]="da" class="value flex-lable-value"><ng-content></ng-content></div>
    <button [attr.da]="daButton" mat-icon-button class="mat-icon-button" (click)=handleClick()>
      <mat-icon>{{editIcon}}</mat-icon>
    </button>
    </div>
    <div class="label">{{label}}</div>
  `,
  styleUrls: ['field.component.scss'],
})
export class EditableFieldComponent extends FieldComponent {
  @Input() editIcon = 'mode_edit';
  @Input() da: string;
  @Input() daButton: string;
  @Output() onIconClick: EventEmitter<any> = new EventEmitter();

  handleClick() {
    this.onIconClick.emit();
  }
}

@Component({
  selector: 'ign-editable-field-inline',
  template: `
    <div class="label-container">
      <div da="fieldValue" class="value flex-lable-value">
      <span class="label">{{label}}</span>
      <ng-content></ng-content>
      </div>
      <button mat-icon-button class="mat-icon-button" (click)=handleClick()>
      <mat-icon>{{editIcon}}</mat-icon>
      </button>
    </div>
  `,
  styleUrls: ['field.component.scss'],
})
export class EditableFieldInlineComponent extends EditableFieldComponent {}
