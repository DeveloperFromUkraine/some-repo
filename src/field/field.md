*** HTML ***
# Field

## Use
This component displays a view field, consisting of a label and a value. 

## Sample

<mat-tab-group>
    <mat-tab label="Component Sample">
        <div class="tab-height">
            <ign-field label="The Answer">42</ign-field>
        </div></mat-tab>
    <mat-tab label="HTML"><div class="tab-height">
        <table style="width:100%">
            &lt;ign-field label="The Answer"&gt;42&lt;/ign-field&gt;
        </table>
    </div></mat-tab>
</mat-tab-group>

## Content

* The contents of the tag will be displayed as the value.

## Style

* A label must be provided.
* A value pertaining to the label must be provided if user input is not requested.

*** TypeScript *** 

*** ExternalTypeScript ***

*** Imports ***