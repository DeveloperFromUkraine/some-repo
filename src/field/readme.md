# Field Component

This component displays a view field, consisting of a label and a value.

## Usage

A label must be provided. The contents of the tag will be displayed as the value. Example usage:

```html
<ign-field label="The Answer">42</ign-field>
```

# Inline Field Component

This component is similar to the field above but slightly different look and feel. Label will appear first followed by the text value on the same row.

## Usage

A label must be provided. The contents of the tag will be displayed as the value. Example usage:

```html
<ign-field-inline label="The Answer">42</ign-field-inline>
```
# Editable Field Component

his component is similar to the field component but also includes an edit button that emits an event that can be subscribed to.

## Usage

A label must be provided. The contents of the tag will be displayed as the value.
An editIcon can be optionaly provided, if none passed the button will display mat-icon mode_edit as default. Example usage:

```html
<ign-editable-field label="The Answer" (onIconClick)="myFn($event)">42</ign-field>
```

# Editable Inline Field Component

This component is similar to the editable field above but slightly different look and feel. Label will appear first followed by the text value followed by the edit button on the same row.

## Usage

A label must be provided. The contents of the tag will be displayed as the value. Example usage:

```html
<ign-editable-field-inline label="The Answer" editIcon="home" (onIconClick)="myFn($event)">42</ign-field-inline>
```
