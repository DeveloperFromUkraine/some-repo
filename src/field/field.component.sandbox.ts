import { sandboxOf } from 'angular-playground';
import { SharedChildModule } from 'app/modules/shared.module';
import { FieldComponent } from './field.component';

export default sandboxOf(FieldComponent, {
  imports: [SharedChildModule],
  declarations: [],
  providers: [],
  declareComponent: false,
}).add('standalone', {
  template: `<ign-field label="Label">Value</ign-field>`,
  styles: [],
});
