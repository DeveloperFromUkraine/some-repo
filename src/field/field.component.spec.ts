import { ComponentFixture, TestBed } from '@angular/core/testing';
import {FieldComponent, FieldInlineComponent, EditableFieldComponent, EditableFieldInlineComponent} from './field.component';
import { ComponentTest } from '../../test/test-bed/component';
import { By } from '@angular/platform-browser';
import { DebugElement } from '@angular/core';
import { MatIconModule } from '@angular/material';

describe('FieldComponent', () => {
    let fixture: ComponentFixture<FieldComponent>;
    let component: FieldComponent;
    let de: DebugElement;
    let ne: HTMLElement;

    beforeEach(async () => {
        ComponentTest.createTestBed([], [FieldComponent]);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(FieldComponent);
        component = fixture.componentInstance;

        fixture.detectChanges();
    });

    it('should set label input', async () => {
        de = fixture.debugElement.query(By.css('.label'));
        ne = de.nativeElement;

        component.label = 'label!';
        await fixture.detectChanges();

        expect(ne.textContent).toContain(component.label);
    });

    it('should match snapshot', async () => {
        component.label = 'label!';
        await fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });
});

describe('FieldInlineComponent', () => {
    let fixture: ComponentFixture<FieldInlineComponent>;
    let component: FieldInlineComponent;
    let de: DebugElement;
    let ne: HTMLElement;

    beforeEach(async () => {
        ComponentTest.createTestBed([], [FieldInlineComponent]);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(FieldInlineComponent);
        component = fixture.componentInstance;

        fixture.detectChanges();
    });

    it('should set label input', async () => {
        de = fixture.debugElement.query(By.css('.label'));
        ne = de.nativeElement;

        component.label = 'label!';
        await fixture.detectChanges();

        expect(ne.textContent).toContain(component.label);
    });
});

describe('EditableFieldComponent', () => {
    let fixture: ComponentFixture<EditableFieldComponent>;
    let component: EditableFieldComponent;
    let de: DebugElement;
    let ne: HTMLElement;

    beforeEach(async () => {
        ComponentTest.createTestBed([MatIconModule], [EditableFieldComponent]);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(EditableFieldComponent);
        component = fixture.componentInstance;

        fixture.detectChanges();
    });

    it('should set label input', async () => {
        de = fixture.debugElement.query(By.css('.label'));
        ne = de.nativeElement;

        component.label = 'label!';
        await fixture.detectChanges();

        expect(ne.textContent).toContain(component.label);
    });

    it('should show mode_edit icon when none specified', async() => {
        de = fixture.debugElement.query(By.css('mat-icon'));
        ne = de.nativeElement;
        
        const DEFAULT_LABEL =  'mode_edit';
        await fixture.detectChanges();

        expect(ne.textContent).toContain(DEFAULT_LABEL);
    });

    it('should show user defined icon when none specified', async() => {
        de = fixture.debugElement.query(By.css('mat-icon'));
        ne = de.nativeElement;
        
        component.editIcon = 'home';
        await fixture.detectChanges();

        expect(ne.textContent).toContain(component.editIcon);
    });

    it('should call handleClick when click event emitted', async () => {
        de = fixture.debugElement.query(By.css('.mat-icon-button'));
        ne = de.nativeElement;

        let spy = jest.fn();

        component.onIconClick.subscribe(spy);
        de.triggerEventHandler('click', null);
        await fixture.detectChanges();

        expect(spy).toHaveBeenCalled();
    });
});
