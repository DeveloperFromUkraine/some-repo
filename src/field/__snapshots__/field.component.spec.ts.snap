// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableFieldComponent should match snapshot 1`] = `
<ign-editable-field
  editIcon={[Function String]}
  label={[Function String]}
  onIconClick={[Function EventEmitter]}
>
  
      <div
    class="label-container"
  >
    
      
    <div
      class="value flex-lable-value"
      da="fieldValue"
    />
    
      
    <button
      class="mat-icon-button"
      mat-icon-button=""
    >
      
        
      <mat-icon
        aria-hidden="true"
        class="mat-icon material-icons"
        role="img"
      >
        mode_edit
      </mat-icon>
      
      
    </button>
    
      
  </div>
      <div
    class="label"
  >
    label!
  </div>
    
</ign-editable-field>
`;

exports[`EditableFieldInlineComponent should match snapshot 1`] = `
<ign-editable-field-inline
  editIcon={[Function String]}
  label={[Function String]}
  onIconClick={[Function EventEmitter]}
>
  
      <div
    class="label-container"
  >
    
        
    <div
      class="value flex-lable-value"
      da="fieldValue"
    >
      
        
      <span
        class="label"
      >
        label!
      </span>
      
        
      
        
    </div>
    
        
    <button
      class="mat-icon-button"
      mat-icon-button=""
    >
      
        
      <mat-icon
        aria-hidden="true"
        class="mat-icon material-icons"
        role="img"
      >
        mode_edit
      </mat-icon>
      
        
    </button>
    
      
  </div>
    
</ign-editable-field-inline>
`;

exports[`Field should match snapshot 1`] = `
<ign-field
  label={[Function String]}
>
  
      <div
    class="value"
    da="fieldValue"
  />
      <div
    class="label"
  >
    label!
  </div>
    
</ign-field>
`;

exports[`FieldInlineComponent should match snapshot 1`] = `
<ign-field-inline
  label={[Function String]}
>
  
      <div
    class="value"
    da="fieldValue"
  >
    
      
    <span
      class="label"
    >
      label!
    </span>
    
      
    
      
  </div>
    
</ign-field-inline>
`;
